java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/protocol/ServerTest.java

```diff
@@ -3228,7 +3228,12 @@ public CompletableFuture<ApplyWorkspaceEditResponse> applyEdit(ApplyWorkspaceEdi
 
             @Override
             public CompletableFuture<List<QuickPickItem>> showQuickPick(ShowQuickPickParams params) {
-                return CompletableFuture.completedFuture(params.getItems().size() > 2 ? params.getItems().subList(0, 2) : params.getItems());
+                List<QuickPickItem> selection = params.getItems().stream()
+                        .filter((i) -> i.getLabel().equals("f1.chars(): IntStream") || i.getLabel().equals("f1.codePoints(): IntStream"))
+                        .sorted((i1, i2) -> i1.getLabel().compareTo(i2.getLabel()))
+                        .collect(Collectors.toList());
+                assertEquals(2, selection.size());
+                return CompletableFuture.completedFuture(selection);
```

Comment:
Hi @sdedic, I have a question.
`ServerTest.testSourceActionDelegateMethod()` assumes that the items are served in a certain order. This PR here however adds an optimization which would return a random order due to the fact that `ElementUtilities.getMembers()` uses HashMaps internally.

The problem is that I don't know the requirements.
 - is it ok to change this test
 - or does getMembers() have to return a certain order
 - or does the LSP code have to sort the results somewhere, so the correct way would be to revert this test and fix it in LSP code

interestingly this was the only test which fails. NB itself does display the choices in the same sorted order as before too.

Reply:
@mbien AFAIK `ElementUtilities.getMembers` is typed as List (= ordered) and so it is javac's `Elements.getAllMembers` this one relies on. Javac internally uses something (ScopeImpl) that seems like LinkedHashMap so iteration happens in the insert order = lexical order.

Reply:
insert order == lexical order ... doesn't make a lot of sense.  lexical order != scope order != insert order.  it's easy enough to change it to use a linked hashtable I suppose - it's what i was going to use until i saw the already existing insertifnotpresent function that used a plain hashtable. it always sorts alphabetically for presentation anyway, but the scope order for identical names has not changed.

Reply:
@notzed sorry, I don't understand - what `insertifnotpresent` ?

Reply:
I don't think `List` strongly implies ordered (the non-javac `ElementUtilities` returns Iterable btw).

We have to figure out what order it is supposed to be and then write it down in the doc so that nobody has to guess again :)

Further if we decide that the order is important, we have to add a test for this. Its not good that one single high level test which is completely unrelated to the given utility methods fails, everything else is green - including a working IDE.

If we decide that the order is not important, either the test has to be fixed as proposed here, or the LSP code has to sort things somewhere to make the test work.

would be good to have input from @jlahoda for this but I completely understand if he can't since he is busy with language JEPs.

Reply:
@sdedic @notzed I switched to LinkedHashMap since this might make it easier to integrate this without doing more research. Completion is still as fast as before far as I can tell.

@sdedic should we keep the change in `ServerTest` or should I revert it (I reverted it for now just to proof that everything is working like before). Assuming everything is green - would be great if you could review this PR, or maybe you know someone who would want to review this :)

would be good to merge this some time soon so that it gets some testing in before release if there are no concerns.

Reply:
Please keep the changes in the test. The importance of the order **is not mentioned** in the Javadoc anywhere and as you noted, E.U.  returns `Iterable` = no guarantees. It's good not to break the de-facto behaviour (`LinkedHashMap`), but the test should stick to the API.

Reply:
@sdedic understood. Travis is green which indicates that the LinkedMaps work, going to remove the revert-commit again, so that the ServerTest changes are restored.